contracts/helper/RebalancingSetEthBidder.sol

```diff
@@ -0,0 +1,236 @@
+     * @param _transferProxy            The address of TransferProxy
+     * @param _weth                     The address of Wrapped Ether
+     */
+    constructor(
+        IRebalanceAuctionModule _rebalanceAuctionModule,
+        ITransferProxy _transferProxy,
+        IWETH _weth
+    )
+        public
+    {
+        // Commit the RebalanceAuctionModule instance
+        rebalanceAuctionModule = _rebalanceAuctionModule;
+
+        // Commit the TransferProxy instance
+        transferProxy = _transferProxy;
+
+        // Commit the WETH instance
+        weth = _weth;
+
+        // Add approvals of Wrapped Ether to the Transfer Proxy
+        ERC20Wrapper.approve(
+            address(_weth),
+            address(_transferProxy),
+            CommonMath.maxUInt256()
+        );
+    }
+
+    /**
+     * Fallback function. Disallows ether to be sent to this contract without data except when
+     * unwrapping WETH.
+     */
+    function ()
+        external
+        payable
+    {
+        require(
+            msg.sender == address(weth),
+            "RebalancingSetEthBidder.fallback: Cannot receive ETH directly unless unwrapping WETH"
+        );
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Bid on rebalancing a given quantity of sets held by a rebalancing token wrapping or unwrapping
+     * any ETH involved. The tokens are returned to the user.
+     *
+     * @param  _rebalancingSetToken    Address of the rebalancing token being bid on
+     * @param  _quantity               Number of currentSets to rebalance
+     * @param  _allowPartialFill       Set to true if want to partially fill bid when quantity
+                                       is greater than currentRemainingSets
+     */
+
+    function bidAndWithdrawWithEther(
+        address _rebalancingSetToken,
+        uint256 _quantity,
+        bool _allowPartialFill
+    )
+        external
+        payable
+        nonReentrant
+    {
+        // Wrap all Ether sent to the contract
+        weth.deposit.value(msg.value)();
+
+        // Get token addresses
+        address[] memory combinedTokenArray = IRebalancingSetToken(_rebalancingSetToken).getCombinedTokenArray();
+
+        // Get inflow and outflow arrays for the given bid quantity
+        uint256[] memory inflowArray;
+        uint256[] memory outflowArray;
+        (
+            inflowArray,
+            outflowArray
+        ) = IRebalancingSetToken(_rebalancingSetToken).getBidPrice(_quantity);
+
+        // Ensure allowances and transfer non-weth tokens from user
+        depositComponents(
+            combinedTokenArray,
+            inflowArray
+        );
+
+        // Bid in auction
+        rebalanceAuctionModule.bidAndWithdraw(
+            _rebalancingSetToken,
+            _quantity,
+            _allowPartialFill
+        );
+
+        // Withdraw non-weth tokens to user
+        withdrawComponentsToSender(
```

Comment:
I don't like the name of this function as it has the side-effect of not transfering WETH out.

It could be `withdrawNonWethComponentsToSender` to be more clear